plugins/ksp/testData/api/resolveJavaType.kt

```diff
@@ -0,0 +1,32 @@
+// TEST PROCESSOR: ResolveJavaTypeProcessor
+// EXPECTED:
+// kotlin.Int
+// kotlin.String
+// kotlin.IntArray
+// P
+// Bar
+// kotlin.Array<Bar>
+// END
+// FILE: a.kt
+annotation class Test
+@Test
+class Foo<P>: C<P>() {
+
+}
+
+class Bar
+
+// FILE: C.java
+class C<T> {
+    public int intFun() {}
+
+    public String strFun() {}
+
+    public int[] intArrayFun() {}
+
+    public P genericFun() {}
+
+    public Bar BarFun() {}
+
+    public Bar[] BarArryFun() {}
+}
```

Comment:
Can you add some tests for generic types and collections? such as `List<? super Double>`.